bpfd/src/bpf.rs

```diff
@@ -179,19 +197,18 @@ impl BpfManager {
             read(program.data().path.clone()).await?
         };
 
+        // This load is just to verify the Section Name is valid.
+        // The actual load is performed in the XDP or TC logic.
         let mut ext_loader = BpfLoader::new()
             .extension(&program.data().section_name)
             .map_pin_path(map_pin_path.clone())
             .load(&program_bytes)?;
 
         match ext_loader.program_mut(&program.data().section_name) {
-            Some(_) => Ok(()),
-            None => {
-                let _ = fs::remove_dir_all(map_pin_path).await;
-                Err(BpfdError::SectionNameNotValid(
-                    program.data().section_name.clone(),
-                ))
-            }
+            Some(_) => Ok(()), // BILLY: Do we need an unload here?
```

Comment:
No.. HEre we're just loading the program into AYA not actuall loading the program into the kernel to do that you also have to do 

```
    let ext: &mut Extension = bpf
        .program_mut(&v.info.metadata.name)
        .ok_or_else(|| BpfdError::SectionNameNotValid(v.info.metadata.name.clone()))?
        .try_into()?;

    let target_fn = format!("prog{i}");

    ext.load(dispatcher.fd().unwrap(), &target_fn)?;
```

Reply:
see https://github.com/aya-rs/aya/blob/main/aya/src/bpf.rs#L377

vs 

https://github.com/aya-rs/aya/blob/main/aya/src/programs/extension.rs#L71